### PR TITLE
Send cheeps

### PR DESCRIPTION
**Pull request by SnowyPeax**
See this issue for details https://github.com/ITU-BDSA23-GROUP5/Chirp/issues/95

### CODE REVIEW

**Review by LavenderPlush — CHANGES_REQUESTED** [review]
The overall changes seem sensible. It is however strongly advised against in the repository pattern, to create dependencies between repositories. I see that the CheepRepository creates an instance of the AuthorRepository. I would instead advice that you use the context directly, as is otherwise done in the repositories, to access the author table.

**Comment by Philipnah** [comment]
> The overall changes seem sensible. It is however strongly advised against in the repository pattern, to create dependencies between repositories. I see that the CheepRepository creates an instance of the AuthorRepository. I would instead advice that you use the context directly, as is otherwise done in the repositories, to access the author table.

Sounds reasonable! Can you supply a source for this claim?

**Comment by LavenderPlush** [comment]
> > The overall changes seem sensible. It is however strongly advised against in the repository pattern, to create dependencies between repositories. I see that the CheepRepository creates an instance of the AuthorRepository. I would instead advice that you use the context directly, as is otherwise done in the repositories, to access the author table.
> 
> Sounds reasonable! Can you supply a source for this claim?

First and foremost it's always a bad idea to add unnecessary coupling. In this situation I would even say that the business logic should be separated and the call to create a new author shouldn't happen from within the Tweet repository - this currently violates the single responsibility principle from the SOLID principles.
It's hard to find a specific example, but I can refer to the model used in [this article](https://learn.microsoft.com/en-us/dotnet/architecture/microservices/microservice-ddd-cqrs-patterns/infrastructure-persistence-layer-design).

I understand if this isn't very satisfying and I can recommend we ask Helge or Rasmus.

File: test/Chirp.Infrastructure.Tests/CheepRepositoryUnitTests.cs [hunk 1]
```diff
@@ -31,10 +34,43 @@ public CheepRepositoryUnitTests()
...
+		CreateCheepDTO cheep = new CreateCheepDTO()
+		{
+			CheepGuid = new Guid(),
+			Text = cheepMessage,
+			Name = name,
+			Email = email
+		};
+
+		Exception? exception = null;
+
```
Comment (JakobArnfred) [inline test/Chirp.Infrastructure.Tests/CheepRepositoryUnitTests.cs:59]: Many great things seems to be accomplished in this pull request. Maybe in CreateNewCheep_FromNonExistingAuthor_ThrowsException() test, we should not create an exception with null value, and instead move it and the Assert-functions into a "finally"-scope

**Pull request merged**